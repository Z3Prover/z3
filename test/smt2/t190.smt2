
; Copyright (c) 2015 Microsoft Corporation
(set-option :pp.max-depth 100)
(declare-fun p (Int) Bool)
(declare-fun f (Int Real) Bool)
(declare-fun q (Int Bool) Bool)
(declare-fun a () Int)

(push)
(assert (= (ite (> (ite (! (p 0) :lblneg baa) 20 30) a) 0 1) a))
(apply snf)
(echo "----")
(pop)

(push)
(assert (= (ite (> (ite (! (p 0) :lblpos baa) 20 30) a) 0 1) a))
(apply snf)
(echo "----")
(pop)
