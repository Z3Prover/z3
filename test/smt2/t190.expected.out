(goals
(goal
  (= (ite (> (ite z3name!0 20 30) a) 0 1) a)
  (or (not z3name!0) (p 0))
  (or z3name!0 (and (not (p 0)) (! true :lblpos baa)))
  :precision precise :depth 1)
)
----
(goals
(goal
  (= (ite (> (ite z3name!1 20 30) a) 0 1) a)
  (or (not z3name!1) (and (p 0) (! true :lblpos baa)))
  (or z3name!1 (not (p 0)))
  :precision precise :depth 1)
)
----
